source/cellular_bg770_api.c

```diff
@@ -6475,3 +6500,787 @@ CellularError_t Cellular_SetServiceSelection( CellularHandle_t cellularHandle,
+            };
+
+        memcpy(&frequencyBandsCopy, pFrequencyBands, sizeof( frequencyBandsCopy ) );
+
+        /* Limit to frequency bands supported by Quectel. */
+
+        if( !_filterLTEBandMask( &frequencyBandsCopy, &SUPPORTED_BG770_LTE_BAND_MASK, &bandsFiltered ) )
+        {
+            LogError( ( "Cellular_SetLTEFrequencyBands : Failed to filter frequency bands to BG770 supported" ) );
+            cellularStatus = CELLULAR_INTERNAL_FAILURE;
+        }
+
+        if( cellularStatus == CELLULAR_SUCCESS )
+        {
+            if( !_isLTEBandMaskNonZero( pFrequencyBands ) )
+            {
+                LogError( ( "Cellular_SetLTEFrequencyBands : Specified frequency bands not supported by BG770" ) );
+            }
+            else
+            {
+                if( bandsFiltered )
+                {
+                    LogWarn( ( "Cellular_SetLTEFrequencyBands : Unsupported LTE frequency bands removed" ) );
+                }
+            }
+        }
+
+        /* Form the AT command. */
+
+        if( cellularStatus == CELLULAR_SUCCESS )
+        {
+            cellularStatus = Cellular_ConvertLTEBandMaskToHexString( &frequencyBandsCopy,
+                                                                     frequencyBandsBuffer,
+                                                                     sizeof( frequencyBandsBuffer ) );
+            if( cellularStatus != CELLULAR_SUCCESS )
+            {
+                LogError( ( "Cellular_SetLTEFrequencyBands : Failed to convert LTE frequency bands to hex string" ) );
+                cellularStatus = CELLULAR_INTERNAL_FAILURE;
+            }
+        }
+
+        if( cellularStatus == CELLULAR_SUCCESS )
+        {
+            /* MISRA Ref 21.6.1 [Use of snprintf] */
+            /* More details at: https://github.com/FreeRTOS/FreeRTOS-Cellular-Interface/blob/main/MISRA.md#rule-216 */
+            /* coverity[misra_c_2012_rule_21_6_violation]. */
+            ( void ) snprintf( cmdBuf, CELLULAR_AT_CMD_MAX_SIZE, "AT+QCFG=\"band\",0,0x%s,0",
+                               frequencyBandsBuffer );
+            LogDebug( ( "Cellular_SetLTEFrequencyBands: Set Frequency Band command: %s", cmdBuf ) );
+            pktStatus = _Cellular_AtcmdRequestWithCallback( pContext, atReqSetFrequencyBands );
+
+            if( pktStatus != CELLULAR_PKT_STATUS_OK )
+            {
+                LogError( ( "Cellular_SetLTEFrequencyBands: Couldn't send Set Frequency Bands, err: %d", pktStatus ) );
+                cellularStatus = _Cellular_TranslatePktStatus( pktStatus );
+            }
+        }
+    }
+
+    return cellularStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool _cstringToLowercase(char *const cstring)
+{
+    if( cstring == NULL )
+    {
+        return false;
+    }
+
+    for( int i = 0; cstring[ i ] != '\0'; i++ )
+    {
+        cstring[ i ] = tolower( ( int ) cstring[ i ] );
+    }
+
+    return true;
+}

```

Comment:
Why not `string.h`'s `strlwr()`?

Reply:
strlwr() is non-ANSI standard and therefore I stuck with code that is standard.